src/main/java/com/redhat/devtools/intellij/tektoncd/actions/StartAction.java

```diff
@@ -60,18 +62,32 @@ public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object s
                 logger.error("Error: " + e.getLocalizedMessage());
                 return;
             }
-            String finalConfiguration = configuration;
-            StartDialog stdialog = UIHelper.executeInUI(() -> {
-                StartDialog dialog = new StartDialog(null, finalConfiguration, resources);
-                dialog.show();
-                return dialog;
-            });
-            if (stdialog.isOK()) {
+
+            StartResourceModel model = new StartResourceModel(configuration, resources);
+
+            if (!model.isValid()) {
+                UIHelper.executeInUI(() -> Messages.showErrorDialog(model.getErrorMessage(), "Error"));
+                return;
+            }
+
+            boolean noInputsAndOuputs = model.getInputs() == null && model.getOutputs() == null;
+            StartDialog stdialog = null;
+
+            if (!noInputsAndOuputs) {
+                stdialog = UIHelper.executeInUI(() -> {
+                    StartDialog dialog = new StartDialog(null, model.getNamespace(), model.getName(), model.getKind(), model.getInputs(), model.getOutputs(), resources);
+                    dialog.show();
+                    return dialog;
+                });
+            }
+            if (noInputsAndOuputs || stdialog.isOK()) {
                 try {
+                    Map<String, String> params = stdialog == null ? null : stdialog.getParameters();
```

Comment:
In order to respect the MVC model, dialog should be given the model and modify it so once dialog is closed you should work on the model

Reply:
postponed to #53